crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -408,8 +437,13 @@ impl LogletConfiguration {
         match self {
             #[cfg(feature = "replicated-loglet")]
             LogletConfiguration::Replicated(configuration) => {
-                // todo check also whether we can improve the nodeset based on the observed cluster state
-                !observed_cluster_state.is_node_alive(configuration.sequencer)
+                // todo: check whether we can improve the nodeset selection even if all nodes are alive
+                let sequencer_alive = observed_cluster_state.is_node_alive(configuration.sequencer);
+                let all_log_servers_alive = configuration
+                    .nodeset
+                    .iter()
+                    .all(|node_id| observed_cluster_state.is_node_alive(*node_id));
```

Comment:
I think we should try to reconfigure if either the sequencer is dead or if we have lost too many of our log servers and there are spare ones to improve the situation. Only checking for `all_log_server_alive` will mean that we try to reconfigure in a 3 node setup if we lose a single node but cannot improve the situation because there are no spare nodes. In this case, the seal operation happens w/o a reason.

Reply:
Ha! Spot on; I had this and `build_new_replicated_loglet_configuration` exactly backwards - this one suggests reconfiguration when we can't improve, and building a configuration would return None when it should have reused the old config, when we couldn't improve on it. Thank you for this.